src/Agent.Worker/TestResults/TrxResultReader.cs

```diff
@@ -103,6 +101,20 @@ public TestRunData ReadResults(IExecutionContext executionContext, string filePa
                 releaseEnvironmentUri: runContext.ReleaseEnvironmentUri
                 );
 
+            //Parse the Deployment node for the runDeploymentRoot - this is the attachment root. 
+            XmlNode deploymentNode = doc.SelectSingleNode("/TestRun/TestSettings/Deployment");
+            var _attachmentRoot = string.Empty;
+            if (deploymentNode != null && deploymentNode.Attributes["runDeploymentRoot"] != null )
+            {
+                _attachmentRoot = deploymentNode.Attributes["runDeploymentRoot"].Value;
+            }
+            else
+            {
+                _attachmentRoot = Path.GetFileNameWithoutExtension(filePath);
+            }
+            _attachmentLocation = Path.Combine(Path.GetDirectoryName(filePath), _attachmentRoot, "In");
+            _executionContext.Debug(string.Format(CultureInfo.InvariantCulture, "Attachment location: {0}", _attachmentLocation));
```

Comment:
Can we add tests for the same? https://github.com/Microsoft/vsts-agent/blob/master/src/Test/L0/Worker/TestResults/TrxResultReaderTests.cs